test/gateway/eip20_gateway/progress_unstake_with_proof.js

```diff
@@ -0,0 +1,511 @@
+      unstakeMessage.unstakeAccount,
+      `Redeemer address ${eventData._redeemer} from event must be equal to ${unstakeMessage.unstakeAccount}.`,
+    );
+    assert.strictEqual(
+      eventData._beneficiary,
+      unstakeRequest.beneficiary,
+      `Beneficiary address ${eventData._beneficiary} from event must be equal to ${unstakeRequest.beneficiary}.`,
+    );
+    assert.strictEqual(
+      unstakeRequest.amount.eq(eventData._redeemAmount),
+      true,
+      `Redeem amount ${eventData._redeemAmount.toString(10)} from event must be equal to ${unstakeRequest.amount.toString(10)}.`,
+    );
+    assert.strictEqual(
+      eventData._rewardAmount.add(eventData._unstakeAmount).eq(unstakeRequest.amount),
+      true,
+      `Total unstake amount should be equal to sum of reward amount plus unstaked amount to beneficiary`,
+    );
+    assert.strictEqual(
+      eventData._proofProgress,
+      true,
+      'Proof progress from event must be equal to true.',
+    );
+    assert.strictEqual(
+      eventData._unlockSecret,
+      ZeroBytes,
+      `Unlock secret ${eventData._unlockSecret} from event must be equal to ${ZeroBytes}.`,
+    );
+
+  });
+
+  it('should unstake token to the beneficiary address', async function () {
+
+    await gateway.setInboxStatus(unstakeMessage.messageHash, MessageStatusEnum.Declared);
+    let blockHeight = await setStateRoot(gateway);
+    let storageProof = StubData.co_gateway.redeem.proof_data.storageProof[0].serializedProof;
+
+    stakeVaultAddress = await setup(unstakeMessage, gateway, unstakeRequest, stakeVaultAddress, mockToken, accounts);
+    let initialBeneficiaryBalance = await mockToken.balanceOf(unstakeRequest.beneficiary);
+    let initialStakeVaultBalance = await mockToken.balanceOf(stakeVaultAddress);
+
+    await gateway.progressUnstakeWithProof(
+      unstakeMessage.messageHash,
+      storageProof,
+      blockHeight,
+      MessageStatusEnum.Declared
+    );
+
+    let finalBeneficiaryBalance = await mockToken.balanceOf(unstakeRequest.beneficiary);
+    let finalStakeVaultBalance = await mockToken.balanceOf(stakeVaultAddress);
+    let reward = unstakeMessage.gasPrice.mul(unstakeMessage.gasLimit);
+    let expectedBeneficiaryBalance = unstakeRequest.amount.sub(reward);
+
+    assert.strictEqual(
+      finalBeneficiaryBalance.eq(expectedBeneficiaryBalance),
+      true,
+      `Beneficiary balance ${finalBeneficiaryBalance.toNumber(10)}` +
+      `must be equal to ${initialBeneficiaryBalance.add(expectedBeneficiaryBalance).toNumber(10)}.`,
+    );
+    assert.strictEqual(
+      finalStakeVaultBalance.eq(initialStakeVaultBalance.sub(unstakeRequest.amount)),
+      true,
+      `Stake vault balance ${finalStakeVaultBalance.toNumber(10)} ` +
+      `must be equal to ${initialStakeVaultBalance.sub(unstakeRequest.amount).toNumber(10)}.`,
+    );
+
+  });
+
+  it('should reward token to the facilitator address with maximum' +
+    ' possible reward', async function () {
```

Comment:
What does "maximum possible reward" mean here?

Reply:
`gasPrice * gasLimit*, I have added this to it statement.